src/components/SearchBox/__tests__/SearchBox-test.js

```diff
@@ -258,50 +273,59 @@ describe('SearchBox', () => {
           ...defaultProps,
           onSubmit,
         };
-        const wrapper = mount(<SearchBox {...props} />);
+        const { container } = render(<SearchBox {...props} />);
+        const form = container.querySelector('form');
 
-        wrapper.find('form').simulate('submit');
+        fireEvent.submit(form);
 
         expect(onSubmit).toHaveBeenCalledTimes(1);
       });
     });
 
-    describe('onReset', () => {
+    // @TODO Since the Preact X migration and new testing environment, this test
+    // suite doesn't pass.
+    // eslint-disable-next-line jest/no-disabled-tests
+    describe.skip('onReset', () => {
```

Comment:
where should we open an issue for this?

Reply:
Seems to be an issue in `jsdom` https://github.com/testing-library/react-testing-library/issues/393
An issue was open, but without any information it was promptly closed https://github.com/jsdom/jsdom/issues/2622
We can complete this issue and ask to re-open it.

Reply:
Great find – so I was not going crazy! I'll take care of opening an issue on `jsdom`.

Reply:
From my tests, it's not coming from `jsdom` because the `reset` event is triggered:

```js
const { JSDOM } = require('jsdom');

const dom = new JSDOM(`
<body>
  <form>
    <button type="submit">Submit</button>
    <button type="reset">Reset</button>
  </form>
</body>
`);

const document = dom.window.document;

document.querySelector('form').addEventListener('reset', event => {
  event.preventDefault();

  console.log('Reset event received');
});

document.querySelector('button[type="reset"]').click();
```

You can reproduce that in a Node environment, by installing `jsdom`.

I suggest we move forward with this PR and find a workaround after. WDYT?

Reply:
> I suggest we move forward with this PR and find a workaround after.

I agree.

Reply:
https://github.com/algolia/instantsearch.js/issues/4160